features/TESTING.md

```diff
@@ -0,0 +1,80 @@
+# How to contribute to tests
+
+## Unit Tests
+
+Unit tests must be local to the various features, and have no external dependencies.
+Unit tests are meant to be cheap and fast, for this reason we are not providing a way to execute them individually.
+
+Executing `make unittests` will go through all the features and execute all the tests.
+
+## End to end tests
+
+End to end tests must be added to the functests suite that can be found [here](./functests).
+
+All the code testing the same feature must belong to the same subpackage (see [sctp](./functests/sctp)).
+
+In order to allow ginkgo to find and execute the relevant tests, the package must be imported anonymously in the [test_suite_test.go file](./functests/test_suite_test.go) as per the following example:
+
+``` go
+_ "github.com/openshift-kni/baremetal-deploy/features/functests/sctp"
+```
+
+If the tests related to a given feature are intended to run by default when running the suite, they must be added to the `FEATURES` variable assignement in the [Makefile](./Makefile).
+
+### Note
+
+The name of the test entry point must reflect the name of the feature folder under [features](./features). This will make adding new jobs easier.
+
+Each test must not perform any kind of setup, and test the feature assuming that it's already avaliable in the cluster we are testing against.
+
+## Running the e2e tests for a given set of features
+
+This can be done by running a command like
+
+```bash
+FEATURES="sctp ptp" && make functests
```

Comment:
depending on what you want to do, the "&&" is redundant here

Reply:
Ah, nice. Thanks!